src/openms/source/ANALYSIS/ID/SiriusMSConverter.cpp

```diff
@@ -0,0 +1,309 @@
+        }
+        if (p == IonSource::Polarity::NEGATIVE && precursor_charge == -1)
+        {
+          int_charge = -1;
+        }
+        if (p == IonSource::Polarity::POSITIVE && precursor_charge == 0)
+        {
+          int_charge = +1;
+          count_to_pos = count_to_pos + 1;
+        }
+        if (p == IonSource::Polarity::NEGATIVE && precursor_charge == 0)
+        {
+          int_charge = -1;
+          count_to_neg = count_to_neg +1;
+        }
+
+        //get m/z and intensity of precursor != MS1 spectrum
+        double precursor_mz = precursor[0].getMZ();
+        float precursor_int = precursor[0].getIntensity();
+
+        //find corresponding ms1 spectra (precursor)
+        PeakMap::ConstIterator s_it2 = spectra.getPrecursorSpectrum(s_it);
+
+        double test_mz = precursor_mz;
+
+        vector<Peak1D> isotopes;
+        isotopes.clear();
+
+        if (s_it2->getMSLevel() != 1)
+        {
+          count_no_ms1 = count_no_ms1 +1;
+        }
+        //get the precursor in the ms1 spectrum (highest intensity in the range of the precursor mz +- 0.1 Da)
+        else
+        {
+          const MSSpectrum<Peak1D>& spectrum1 = *s_it2;
+
+          // 0.2 Da left and right of the precursor m/z - we do not expect metabolites with charge 5 or higher.
+          Int mono_index = getHighestIntensityPeakInMZRange(test_mz, spectrum1, 0.2, 0.2);
+
+          if (mono_index != -1)
+          {
+            const Peak1D& max_mono_peak = spectrum1[mono_index];
+            isotopes.push_back(max_mono_peak);
+
+            // make sure the 13C isotopic peak is picked up by doubling the (fractional) mass difference (approx. 1.0066)
+            const double C13_dd = 2.0 * (Constants::C13C12_MASSDIFF_U - 1.0);
+            Int iso1_index = getHighestIntensityPeakInMZRange(max_mono_peak.getMZ() + Constants::C13C12_MASSDIFF_U, spectrum1, 0, C13_dd);
+
+            if (iso1_index != -1)
+            {
+              const Peak1D& iso1_peak = spectrum1[iso1_index];
+              isotopes.push_back(iso1_peak);
+              Int iso2_index = getHighestIntensityPeakInMZRange(iso1_peak.getMZ() + Constants::C13C12_MASSDIFF_U, spectrum1, 0, C13_dd);
+
+              if (iso2_index != -1)
+              {
+                const Peak1D& iso2_peak = spectrum1[iso2_index];
+                isotopes.push_back(iso2_peak);
+              }
+            }
+          }
+        }
+
+        String query_id = String("unknown") + String(scan_index);
+
+        if (count == 0) // one ms file
+        {
+          // store data
+          String unique_name =  String(File::getUniqueName()).toQString(); // generate unique name once
+          String tmp_filename = QDir::toNativeSeparators(String(File::getTempDirectory()).toQString()) + "/" + unique_name.toQString() + "_" + query_id.toQString() + ".ms";
+          msfile = tmp_filename;
+
+          // create temporary input file (.ms)
+          os.open(tmp_filename.c_str());
+          if (not os)
```

Comment:
not keyword as replacement for ! doesn't seem to be supported on windows without including iso646.h. I would change it to !.